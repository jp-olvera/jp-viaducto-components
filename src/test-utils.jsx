import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
