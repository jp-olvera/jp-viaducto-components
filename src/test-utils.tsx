import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
