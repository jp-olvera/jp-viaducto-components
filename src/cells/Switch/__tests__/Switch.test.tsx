/* eslint-env jest */

import React from 'react';

import { render, screen, fireEvent } from '../../../test-utils';
import { Switch } from '..';

describe('<Switch/>', () => {
  test('should render properly', () => {
    render(<Switch id='Switch' inputSize='large' />);
    expect(screen.getByTestId('Switch')).toBeVisible();
  });

  test('should be selected', () => {
    const { getByTestId } = render(<Switch id='Switch' />);
    const input = getByTestId('Switch');
    const selector = input.querySelector('input');
    fireEvent.click(selector || window);
    expect(selector?.checked).toEqual(true);
  });

  test('should be not checked', () => {
    const { getByTestId } = render(<Switch id='switch1' />);
    const input = getByTestId('switch1');
    const selector = input.querySelector('input');
    expect(selector?.checked).toEqual(false);
    fireEvent.click(selector || window);
    expect(selector?.checked).toEqual(true);
  });

  test('should be disabled', () => {
    const { getByTestId } = render(
      <Switch id='Switch1' inputSize='small' disabled />,
    );
    expect(
      getByTestId('Switch1').querySelector('input')?.hasAttribute('disabled'),
    ).toEqual(true);
  });

  test('should not be disabled', () => {
    const { getByTestId } = render(
      <Switch id='Switch1' inputSize='default' disabled={false} />,
    );
    expect(
      getByTestId('Switch1').querySelector('input')?.hasAttribute('disabled'),
    ).toEqual(false);
  });
  test('should fire hover event', () => {
    const { getByTestId, container } = render(
      <Switch disabled={false} circular={false} />,
    );
    fireEvent.mouseOver(getByTestId('slider'));
    expect(container).toMatchSnapshot();
  });
});
