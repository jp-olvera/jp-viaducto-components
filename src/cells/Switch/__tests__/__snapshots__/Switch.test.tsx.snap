// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch/> should fire hover event 1`] = `
<div>
  <label
    for="Switch1"
  >
    Switch
    <div
      class="StyledSwitch__DivWrapper-sc-10hvevw-0 kCfAgd ballena-switch-wrapper"
    >
      <label
        class="StyledSwitch-sc-10hvevw-1 kPDNZJ"
        data-testid="Switch1"
        font-size="md"
        for="Switch1"
      >
        <input
          id="Switch1"
          type="checkbox"
          value="false"
        />
        <span
          class="slider round"
          data-testid="slider"
        />
        <div
          class="Spacer__Box-sc-1ka6dkz-0 hyFaJH"
          direction="horizontal"
        />
        <label
          class="switch-label"
          for="Switch1"
        >
          Hola
        </label>
      </label>
    </div>
    ,
  </label>
</div>
`;
